### PR TITLE
Improve ContextTask documentation

### LINKED ISSUE #208: Improve ContextTask XML comments for docs
The top-level summary and remarks of [ContextTask](https://open-ephys.github.io/onix1-bonsai-docs/api/OpenEphys.Onix1.ContextTask.html) is a bit jargony. Additional description, external links, or dumbing-down might help.

image

``oni.Context`` or ``oni.Frame`` don't seem to be valid cref values.

https://github.com/open-ephys/onix-bonsai-onix1/blob/7ac700fb8a696d4d9f8d97c9d1b85765b9951ca6/OpenEphys.Onix1/ContextTask.cs#L12-L20

Don't repeat Clock as an example for the oni.Frame.Clock and its derivatives.

image

https://github.com/open-ephys/onix-bonsai-onix1/blob/7ac700fb8a696d4d9f8d97c9d1b85765b9951ca6/OpenEphys.Onix1/ContextTask.cs#L121-L125

### PR DESCRIPTION
**Pull request by jonnew**
- More details in class description
- Fixes #208

### CODE REVIEW

**Comment by cjsha** [comment]
Can you confirm this? That these remarks for `ContextTask`'s `BlockReadSize` might more aptly belong to [`StartAcquisition`'s `ReadSize`](https://github.com/open-ephys/onix-bonsai-onix1/blob/6c67f9b0072721a8d0c66384432416fe155149bf/OpenEphys.Onix1/StartAcquisition.cs#L28) (which already has a decent description actually).  
https://github.com/open-ephys/onix-bonsai-onix1/blob/6c67f9b0072721a8d0c66384432416fe155149bf/OpenEphys.Onix1/ContextTask.cs#L409-L416

**Comment by cjsha** [comment]
I decided to keep reading on the `ContextTask` page. There are a bunch of members of contextTask which would might be helpful knowing how/when/where they are set. For instance, it might be worth mentioning that `BlockReadSize` and `BlockWriteSize` are set by `ReadSize` and `WriteSize` properties in `StartAcqusition`.

**Review by cjsha — APPROVED** [review]
In addition to line-by-line comments, I also made a couple more regarding other remarks on the ContextTask.cs file that I want you to read before merging. 

This PR is approved but please go through and consider all the comments - I'm not sure it requires another review cycle after you address them (though you can re-request me if you want)

**Comment by jonnew** [comment]
> Can you confirm this? That these remarks for `ContextTask`'s `BlockReadSize` might more aptly belong to [`StartAcquisition`'s `ReadSize`](https://github.com/open-ephys/onix-bonsai-onix1/blob/6c67f9b0072721a8d0c66384432416fe155149bf/OpenEphys.Onix1/StartAcquisition.cs#L28) (which already has a decent description actually).
> 
> https://github.com/open-ephys/onix-bonsai-onix1/blob/6c67f9b0072721a8d0c66384432416fe155149bf/OpenEphys.Onix1/ContextTask.cs#L409-L416

Good find. It turns out that if we do the following:

```
/// <summary>
/// Gets the number of bytes read per cycle of the <see cref="ContextTask"/>'s acquisition
/// thread.
/// </summary>
/// <inheritdoc cref = "StartAcquisition.ReadSize"/>
public int BlockReadSize => ctx.BlockReadSize;
```
then we override the summary (since this property only has a getter) but we import the `remarks` from `StartAcquisition.ReadSize`. Very useful.

File: OpenEphys.Onix1/ContextTask.cs [hunk 1]
```diff
@@ -10,34 +10,46 @@
...
+    /// specification and API describe a general purpose acquisition system architecture and programming
+    /// interface for communication with a host PC. One requirement of ONI is that a host application must
+    /// hold a "context" that contains handles for hardware communication, data acquisition parameters, etc.
+    /// for a particular hardware controller, such as the ONIX PCIe card. <see cref="ContextTask"/> fulfils
+    /// this role for this library. Additionally, once data acquisition is started by the <see
+    /// cref="StartAcquisition"/> operator, <see cref="ContextTask"/> performs the following:
+    /// <list type="bullet">
+    /// <item><description>It automatically reads and distributes data from hardware using a dedicated acquisition
+    /// thread.</description></item>
+    /// <item><description>It allows data to be written to devices that accept them.</description></item>
```
[the remark this answers is not in the document]
Comment (jonnew) [reply OpenEphys.Onix1/ContextTask.cs:26]: ones reading the other is reading.

**Pull request merged**